tests/bundle/Functional/ResourceAssertionsTrait.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Bundle\Rest\Functional;
+
+use JsonSchema\Validator;
+
+trait ResourceAssertionsTrait
+{
+    final protected static function assertResponseMatchesXmlSnapshot(string $content, ?string $file = null): void
+    {
+        self::assertStringMatchesSnapshot($content, 'xml', $file);
+    }
+
+    final protected static function assertResponseMatchesJsonSnapshot(string $content, ?string $file = null): void
+    {
+        self::assertStringMatchesSnapshot($content, 'json', $file);
+    }
+
+    /**
+     * @phpstan-param "xml"|"json"|null $type
+     */
+    final protected static function assertStringMatchesSnapshot(
+        string $content,
+        ?string $type = null,
+        ?string $file = null
+    ): void {
+        if ($file === null) {
+            $classInfo = new \ReflectionClass(static::class);
+            $class = substr(static::class, strrpos(static::class, '\\') + 1);
+            $classFilename = $classInfo->getFileName();
+            self::assertNotFalse($classFilename);
+            $file = dirname($classFilename) . '/' . $class . '.' . ($type ?? 'log');
+        }
+        if (!file_exists($file)) {
+            file_put_contents($file, rtrim($content, "\n") . "\n");
+        }
```

Comment:
we need here a PHPUnit warning (`addWarning`) that a file has been created with information that this is expected one-time only when trying to add coverage for a new endpoint.

Reply:
Added `IBEXA_REST_GENERATE_SNAPSHOTS` environment variable. Moved default snapshot file location to a `_snapshot` subdirectory.

See efcde2c8.